tests/orchestrator/test_sequential_orchestrator.py

```diff
@@ -20,6 +20,34 @@ def generator_function(context):
 
     return outputs
 
+def generator_function_deterministic_utc_time(context):
+    outputs = []
+
+    now = context.current_utc_datetime
+    if not now:
+        raise Exception("No time! 1st attempt")
+    task1 = yield context.call_activity("Hello", "Tokyo")
+
+    now = context.current_utc_datetime
+    if not now:
+        raise Exception("No time! 2nd attempt")
+    task2 = yield context.call_activity("Hello", "Seattle")
+
+    now = context.current_utc_datetime
+    if not now:
+        raise Exception("No time! 3rd attempt")
+    task3 = yield context.call_activity("Hello", "London")
+
+    now = context.current_utc_datetime
+    if not now:
+        raise Exception("No time! 4th attempt")
+
+    outputs.append(task1)
+    outputs.append(task2)
+    outputs.append(task3)
+
+    return outputs
```

Comment:
This is a good test, but you can make it even better by make sure that the value of `current_utc_datetime` is different for each of the calls you make. I wrote a similar test just the other day for the SQL backend and it uncovered a bunch of issues. Here is a reference to it if you think it might be useful to translate into Python: https://github.com/cgillum/durabletask-sqlserver/blob/a1a308c14b13ffca6a751f9fcdce111d55f40336/test/DurableTask.SqlServer.Tests/Integration/Orchestrations.cs#L114-L142

Reply:
I see. I want to take a step back to sync on the context here though as I think our tests are validating different scenarios.
 
Mine makes sure that, in the case that we cannot distinguish the timestamps of two `OrchestratorStarted` events (maybe they occur to close to one another), `current_utc_datetime` would not be `None`. I was testing this specifically because, only in this scenario, `current_utc_datetime` would get explicitly assigned to `None`.

I think yours is testing that the deterministic time API is correctly updating, now? I do think that's a good test, so I'll add it :) but I just wanted to make sure we were on the same page as to what I was trying to test for.

Reply:
I understand it's beyond the scope of your PR, but given the fact that you decided to name this function `generator_function_deterministic_utc_time` I thought the suggestion might be appropriate. Of course, there would also need to be the `None` checks too since that's ultimately what you're trying to fix.